tactics/equality.ml

```diff
@@ -490,29 +505,34 @@ let general_rewrite_clause l2r with_evars ?tac c cl =
               (general_rewrite ~where:(Some id) ~l2r (occs_of occs) ~freeze:false ~dep:true ~with_evars ?tac c)
               (do_hyps l)
         in
-        if cl.concl_occs == NoOccurrences then do_hyps l else
-          tclTHENFIRST
-            (general_rewrite ~where:None ~l2r (occs_of cl.concl_occs) ~freeze:false ~dep:true ~with_evars ?tac c)
-            (do_hyps l)
+        let tac =
+          if cl.concl_occs == NoOccurrences then do_hyps l
+          else
+            tclTHENFIRST
+              (general_rewrite ~where:None ~l2r (occs_of cl.concl_occs) ~freeze:false ~dep:true ~with_evars ?tac c)
+              (do_hyps l)
+        in
+        begin match l with
+        | [] | [_] ->
```

Comment:
>Could it be that l is empty most of the time? This would incur a cost because check_and_clear_in_constr has no fast-path when the list of ids is empty.

Bench with the [] case added: https://gitlab.com/coq/coq/-/jobs/2018394147